visualization/app/codeCharta/ui/attributeSideBar/attributeSideBarSecondaryMetrics/secondaryMetrics.selector.ts

```diff
@@ -0,0 +1,28 @@
+import { createSelector } from "../../../state/angular-redux/store"
+import { selectedNodeSelector } from "../../../state/selectors/selectedNode.selector"
+import { CcState } from "../../../state/store/store"
+import { primaryMetricNamesSelector } from "../attributeSideBarPrimaryMetrics/primaryMetricNames.selector"
+import { Metric, shouldShowAttributeTypeSelector } from "../util/metricHelper"
+
+export type SecondaryMetric = Metric & {
+	showDeltaValue: boolean
+}
+
+export const secondaryMetricsSelector: (state: CcState) => SecondaryMetric[] = createSelector(
+	[selectedNodeSelector, primaryMetricNamesSelector],
+	(selectedNode, primaryMetricNames) => {
+		if (!selectedNode) return []
+
+		const primaryMetricNamesList = Object.values(primaryMetricNames)
+		const secondaryMetricNames = Object.keys(selectedNode.attributes)
+			.filter(metricName => metricName !== "unary" && !primaryMetricNamesList.includes(metricName))
+			.sort()
```

Comment:
I very well see why you changed it. This does however change the actual behavior. The former behavior gathered all metrics from all nodes and combined them in a list. Now, only the individual node is looked at. That way, if one node contains different metrics than another node, we'll also display other metrics. Before, the unknown metrics were set to zero.

I personally am fine with this change as I can not think of a good use case for inconsistent metrics for now. We should however outline this behavior change in the changelog.

Reply:
Thanks for the question. I doubled checked and came to the following conclusion:
- Previous implementation also took only attributes of current selected node into account
- But edge values where always secondary metrics, even if it is selected as primary metric as well (see screenshot below). I think new behavior is as expected and I would add an entry to change section of changelog. But I would be fine with re-implementing old behavior as well.
![image](https://user-images.githubusercontent.com/3596742/143835168-c03e0eae-0615-4bdd-9b01-d1041e3eaebd.png)


For reference previous implementation:
```ts
private updateSortedMetricKeysWithoutPrimaryMetrics(buildingSelected: boolean) {
		if (this._viewModel.node) {
			const metricValues = new Set(Object.values(this._viewModel.primaryMetricKeys.node))

			const secondaryMetricKeys = Object.keys(this._viewModel.node.attributes)
				.filter(key => !metricValues.has(key))
				.sort()

			const secondaryMetrics = []

			for (const metricKey of secondaryMetricKeys) {
				// Temporary solution to remove unary as viewable metric
				if (metricKey === "unary") {
					continue
				}

				const metricType =
					metricKey in this._viewModel.node.edgeAttributes
						? this.edgeMetricDataService.getAttributeTypeByMetric(metricKey)
						: this.nodeMetricDataService.getAttributeTypeByMetric(metricKey)
				secondaryMetrics.push({ name: metricKey, type: metricType } as SecondaryMetric)
			}
			if (!buildingSelected) {
				this.storeService.dispatch(setSecondaryMetrics(secondaryMetrics))
			}
			this._viewModel.secondaryMetrics = secondaryMetrics
		}
	}
```
with 
```
this._viewModel.primaryMetricKeys := {
                node: {
			area: string
			color: string
			height: string
		}
		edge: {
			edge: string
		}
	}
```

Reply:
Thanks for cross checking! The pictures looks rather odd to me though (both, the old and the new one): the edge metric has no indication if it's a sum or the median. The values also deviate in the old version: as primary value different values are visible than the same entry as secondary value?

Reply:
yes. The meaning isn't clear to me as well. The screenshot is from the default map with local development, which is not a real map, as far as I know. I also checked a real map and it doesn't make sense to me yet. Maybe we need an investigation task for it? I would like to restrict migration PRs to migration.

Reply:
Resolve conversation with: 
- I will adjust the hand crafted cc.json for local development some time
- created #2528